File: ios/Infra/Adapters/Extractable.swift
//
//  Extractable.swift
//  uncompress
//
//  Created by Adriano Souza Costa on 14/03/21.
//

import Foundation

public protocol Extractable {

    var compatibilities: [Compatibility] { get }
    
    func extract(
        _ filePath: URL,
        to destination: URL,
        overwrite: Bool,
        password: String?
    ) throws
    
    func isProtected(_ filePath: URL) throws -> Bool
    
}

File: android/src/main/java/com/uncompress/infra/adapters/Extractable.kt
package com.uncompress.infra.adapters

import com.uncompress.domain.enum.Compatibility
import java.io.IOException

interface Extractable {
  val compatibilities: List<Compatibility>

  @Throws(IOException::class)
  fun extract(
    filePath: String,
    destination: String,
    override: Boolean,
    password: String?
  )
}
